types/pino-http/index.d.ts

```diff
@@ -15,11 +15,15 @@ declare function PinoHttp(stream?: DestinationStream): PinoHttp.HttpLogger;
 declare namespace PinoHttp {
     type HttpLogger = (req: IncomingMessage, res: ServerResponse) => void;
 
+    // When we upgrade to TypeScript 3.0, we can use `unknown`.
+    type PoorMansUnknown = {} | null | undefined;
+
     interface Options extends LoggerOptions {
         logger?: Logger;
         genReqId?: GenReqId;
         useLevel?: Level;
         stream?: DestinationStream;
+        customLogLevel?: (res: ServerResponse, error: PoorMansUnknown) => Level;
```

Comment:
Why isn't `error` just an optional [error](https://nodejs.org/api/stream.html#stream_event_error) `(res: ServerResponse, error?: Error) => Level;`?

Reply:
Good question. Exceptions can be any type, e.g. `throw 1`. I see the docs specify the `error` event will emit an `Error` type, but I wonder if that's true—surely the stream could emit an error of any type, just like exceptions can?

Reply:
Exceptions are not necessarily errors. You **can** throw an Error object or just pass a newly created Error object to a callback function but you don't have to throw it.

You can throw a string or a number but this is not a good idea because it doesn't preserve the call stack. All error callbacks in Node.js give you an [Error](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error)

Pino use streams and streams give you an [Error Object](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/pino/index.d.ts).

Reply:
Updated